x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/SyntheticSourceIndexSettingsProvider.java

```diff
@@ -101,6 +101,18 @@ boolean newIndexHasSyntheticSourceUsage(
 
         try {
             var tmpIndexMetadata = buildIndexMetadataForMapperService(indexName, templateIndexMode, indexTemplateAndCreateRequestSettings);
+            var indexMode = tmpIndexMetadata.getIndexMode();
+            if (SourceFieldMapper.INDEX_MAPPER_SOURCE_MODE_SETTING.exists(tmpIndexMetadata.getSettings())
+                || indexMode == IndexMode.LOGSDB
+                || indexMode == IndexMode.TIME_SERIES) {
+                // in case when index mode is either tsdb or logsdb and only _source.mode attribute is specified, then the default could
+                // be wrong. However, it doesn't really matter, because if it is set to stored, then adding index.mapping.source.mode has
+                // no effect. And _source.mode can't be set to disabled, because that isn't allowed with logsdb/tsdb.
```

Comment:
`However, it doesn't really matter, because if it is set to stored, then adding index.mapping.source.mode has no effect.`

Mind explaining this? What happens if indexMode is LOGSDB and _source.mode is 'stored', if we don't read the latter?

Reply:
I see, failing to inject `stored` here won't make a difference. Maybe clarify in the comment.

Reply:
I updated the comment: cf601211829a95391b2dfc3acbc420bdf6156b97